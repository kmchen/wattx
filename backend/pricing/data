LTC,
XRP,
EOS,
BCH,
ETH,
BTC,
TX,
EL,
HB,
XBTC21,
CLOAK,
LINDA,
YOYOW,
SPANK,
SNGLS,
WAVES,
TRUMP,
USNBT,
WOMEN,
XSPEC,
STORJ,
VOISE,
UCASH,
THETA,
UNIFY,
VIDZ,
TELL,
TOKC,
TIME,
UNIT,
VERI,
VEC2,
UTNP,
UBTC,
VOLT,
VPRC,
WINK,
WISH,
XCXT,
XAUR,
QBIC,
DEEX,
ZENI,
XSTC,
XPTX,
ZEIT,
ZIPT,
NOAH,
MSCN,
BRIA,
HTML,
CMCT,
PLBT,
PIRL,
BTCZ,
EURS,
USDC,
TRST,
TUBE,
AION,
DASC,
BCPT,
ELEC,
NEXO,
CTXC,
QASH,
CCRB,
MANA,
IOST,
DATA,
GUSD,
ECTE,
TUSD,
MITH,
DOGE,
DGTX,
USDT,
DASH,
QTUM,
XLM,
BTG,
NEO,
ETC,
ZEC,
XMR,
ZRX,
REP,
B2G,
BSV,
DAI,
ADA,
OMG,
TRX,
ETP,
BTW,
MGO,
EDO,
ORS,
BNB,
XTZ,
NIO,
ADK,
PZM,
VEE,
BCA,
XEM,
SAN,
GNT,
INT,
LSK,
FUN,
VEX,
VET,
RIF,
GAS,
ODE,
ENJ,
WAX,
PAX,
GNO,
EDG,
BTT,
CPC,
BAT,
AGI,
MNX,
MKR,
BCI,
RLC,
TDS,
EMC,
CCL,
KNC,
DIG,
AVT,
DGX,
XVG,
AUC,
RCN,
BCD,
ATB,
INK,
ELF,
ZIL,
LKK,
LEO,
AID,
DXT,
DIM,
DTH,
DGB,
STQ,
TNB,
MNC,
HBZ,
VRS,
SNT,
TRF,
YOC,
AC3,
CHX,
HOT,
NVC,
PXG,
MLN,
ANT,
CND,
BFT,
EKO,
NEU,
CNN,
MCO,
WIZ,
IPL,
GRS,
KRB,
CVC,
DOR,
NTK,
UQC,
PAY,
SVD,
OXY,
NIM,
REC,
PUT,
TKA,
AMB,
ZYD,
NYC,
XPD,
SPF,
HWC,
ZNY,
ZSC,
ZRC,
ZUR,
ZPT,
ZET,
ZCN,
ZAP,
XZC,
XUC,
XRA,
XQN,
XPY,
XPC,
XOT,
XMO,
XMG,
XJO,
XIN,
XHI,
XFC,
XDN,
XCO,
XBY,
XCT,
XBP,
WRC,
WPR,
WTC,
WIN,
WGR,
WIC,
WGO,
WCT,
WCO,
WBB,
VTA,
VSX,
VTC,
VLT,
VIB,
VIT,
VIA,
UTT,
UTK,
USC,
UNO,
UNI,
UIS,
UFR,
UKG,
TTC,
TSC,
TRC,
TOA,
TNT,
TKN,
TIT,
TIE,
TFL,
TEK,
TCN,

XAV, {Data:map[]}
XCE, {Data:map[]}
XCRE, {Data:map[]}
XBTS, {Data:map[]}
XB, {Data:map[]}
XBTA, {Data:map[]}
XAG, {Data:map[]}
XBG, {Data:map[]}
X2, {Data:map[]}
X2C, {Data:map[]}
XBS, {Data:map[]}
WPC, {Data:map[]}
WTHC, {Data:map[]}
WTT, {Data:map[]}
WRP, {Data:map[]}
WTRY, {Data:map[]}
WRT, {Data:map[]}
WOC2, {Data:map[]}
WOP, {Data:map[]}

{map[] }: {Data:map[]}
WNET, {Data:map[]}
{map[] }: {Data:map[]}
WLC, {Data:map[]}
{map[] }: {Data:map[]}
WOK, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[WOMEN:{Quote:map[USD:{Price:1.9270794e-06}] Symbol:WOMEN}]}
WOCX, {Data:map[]}
{map[] }: {Data:map[]}
WMC, {Data:map[]}
{map[] }: {Data:map[]}
WITCH, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[WISH:{Quote:map[USD:{Price:0.019474557}] Symbol:WISH}]}
WHO, {Data:map[]}
{map[] }: {Data:map[]}
WISC, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[WIN:{Quote:map[USD:{Price:0.00031011467}] Symbol:WIN}]}
{map[] }: {Data:map[WGR:{Quote:map[USD:{Price:0.074267626}] Symbol:WGR}]}
{map[] }: {Data:map[WIC:{Quote:map[USD:{Price:0.00035265798}] Symbol:WIC}]}

WINE, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[WINK:{Quote:map[USD:{Price:0.00023524005}] Symbol:WINK}]}
{map[] }: {Data:map[WGO:{Quote:map[USD:{Price:0.013761423}] Symbol:WGO}]}
WCG, {Data:map[]}
{map[] }: {Data:map[]}
WGC, {Data:map[]}
{map[] }: {Data:map[]}
WCN, {Data:map[]}
{map[] }: {Data:map[]}
WCASH, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[WCT:{Quote:map[USD:{Price:0.4892699}] Symbol:WCT}]}
{map[] }: {Data:map[WCO:{Quote:map[USD:{Price:0.0049899793}] Symbol:WCO}]}
WEEK, {Data:map[]}
{map[] }: {Data:map[]}
WEX, {Data:map[]}
{map[] }: {Data:map[]}
WDESK, {Data:map[]}
{map[] }: {Data:map[]}
WAM, {Data:map[]}
{map[] }: {Data:map[]}
WASH, {Data:map[]}
{map[] }: {Data:map[]}
WAY, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[WBB:{Quote:map[USD:{Price:0.01975689}] Symbol:WBB}]}
WARP, {Data:map[]}
{map[] }: {Data:map[]}
VVI, {Data:map[]}
{map[] }: {Data:map[]}
VUC, {Data:map[]}
{map[] }: {Data:map[]}
VTY, {Data:map[]}
{map[] }: {Data:map[]}
WA, {Data:map[]}
{map[] }: {Data:map[]}
VRP*, {Data:map[]}
{map[] }: {Data:map[]}
VSLICE, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[VTA:{Quote:map[USD:{Price:5.545305e-07}] Symbol:VTA}]}
{map[] }: {Data:map[VSX:{Quote:map[USD:{Price:0.0009785331}] Symbol:VSX}]}
{map[] }: {Data:map[VTC:{Quote:map[USD:{Price:0.5746326}] Symbol:VTC}]}
VTX, {Data:map[]}
{map[] }: {Data:map[]}
VTL, {Data:map[]}
{map[] }: {Data:map[]}
VTN, {Data:map[]}
{map[] }: {Data:map[]}
VSM, {Data:map[]}
{map[] }: {Data:map[]}
VNTX, {Data:map[]}
{map[] }: {Data:map[]}
VLTC, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[VOLT:{Quote:map[USD:{Price:0.00023485861}] Symbol:VOLT}]}
{map[] }: {Data:map[VPRC:{Quote:map[USD:{Price:1.9550732e-06}] Symbol:VPRC}]}
{map[] }: {Data:map[VLT:{Quote:map[USD:{Price:0.0071089272}] Symbol:VLT}]}
VOL, {Data:map[]}
{map[] }: {Data:map[]}
VOYA, {Data:map[]}
{map[] }: {Data:map[]}
VPN, {Data:map[]}
{map[] }: {Data:map[]}
VK, {Data:map[]}
{map[] }: {Data:map[]}
VERS, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[VIB:{Quote:map[USD:{Price:0.027916618}] Symbol:VIB}]}
VIP, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[VIT:{Quote:map[USD:{Price:0.002399928}] Symbol:VIT}]}
{map[] }: {Data:map[VIDZ:{Quote:map[USD:{Price:0.00027408963}] Symbol:VIDZ}]}
VIRAL, {Data:map[]}
{map[] }: {Data:map[]}
VISIO, {Data:map[]}
{map[] }: {Data:map[]}
VIC, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[VIA:{Quote:map[USD:{Price:0.46934038}] Symbol:VIA}]}
VEC, {Data:map[]}
{map[] }: {Data:map[]}
VEN, {Data:map[]}
{map[] }: {Data:map[]}
VEGA*, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[VERI:{Quote:map[USD:{Price:16.828173}] Symbol:VERI}]}
VENE, {Data:map[]}
{map[] }: {Data:map[]}
VCOIN, {Data:map[]}
{map[] }: {Data:map[]}
VERSA, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[VEC2:{Quote:map[USD:{Price:0.00023596066}] Symbol:VEC2}]}
VEG, {Data:map[]}
{map[] }: {Data:map[]}
UTN, {Data:map[]}
{map[] }: {Data:map[]}
VAPOR, {Data:map[]}
{map[] }: {Data:map[]}
UXC, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[UTNP:{Quote:map[USD:{Price:0.0041632745}] Symbol:UTNP}]}
UTLE, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[UTT:{Quote:map[USD:{Price:0.3151384}] Symbol:UTT}]}
U, {Data:map[]}
{map[] }: {Data:map[]}
VAL, {Data:map[]}
{map[] }: {Data:map[]}
VAK, {Data:map[]}
{map[] }: {Data:map[]}
URO, {Data:map[]}
{map[] }: {Data:map[]}
URC, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[UTK:{Quote:map[USD:{Price:0.027198115}] Symbol:UTK}]}
{map[] }: {Data:map[USNBT:{Quote:map[USD:{Price:0.061483845}] Symbol:USNBT}]}
{map[] }: {Data:map[USC:{Quote:map[USD:{Price:0.0036034598}] Symbol:USC}]}
USDET, {Data:map[]}
{map[] }: {Data:map[]}
UTA, {Data:map[]}
{map[] }: {Data:map[]}
US-NBT, {Data:map[]}
{map[] }: {Data:map[]}
USDE, {Data:map[]}
{map[] }: {Data:map[]}
UNC, {Data:map[]}
UNITS, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[]}
UNF, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[UNO:{Quote:map[USD:{Price:96.43203}] Symbol:UNO}]}
{map[] }: {Data:map[UNIFY:{Quote:map[USD:{Price:0.003447291}] Symbol:UNIFY}]}
UNRC, {Data:map[]}
{map[] }: {Data:map[]}
UNY, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[UNIT:{Quote:map[USD:{Price:0.005243793}] Symbol:UNIT}]}
{map[] }: {Data:map[UNI:{Quote:map[USD:{Price:0.0019542812}] Symbol:UNI}]}
UMC, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[UIS:{Quote:map[USD:{Price:0.0012552241}] Symbol:UIS}]}
UCI, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[UFR:{Quote:map[USD:{Price:0.053667884}] Symbol:UFR}]}
UDOWN, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[UKG:{Quote:map[USD:{Price:0.03236931}] Symbol:UKG}]}
UNB, {Data:map[]}
{map[] }: {Data:map[]}
ULTC, {Data:map[]}
{map[] }: {Data:map[]}
UCOM, {Data:map[]}
{map[] }: {Data:map[]}
UBIQ, {Data:map[]}
{map[] }: {Data:map[]}
UB, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TX:{Quote:map[USD:{Price:0.31829596}] Symbol:TX}]}
{map[] }: {Data:map[UBTC:{Quote:map[USD:{Price:1.0705165}] Symbol:UBTC}]}
TWERK, {Data:map[]}
{map[] }: {Data:map[]}
TWLV, {Data:map[]}
{map[] }: {Data:map[]}
UAE, {Data:map[]}
{map[] }: {Data:map[]}
TWO, {Data:map[]}
{map[] }: {Data:map[]}
TWIST, {Data:map[]}
{map[] }: {Data:map[]}
TREE, {Data:map[]}
{map[] }: {Data:map[]}
TSE, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TRUMP:{Quote:map[USD:{Price:0.015751464}] Symbol:TRUMP}]}
{map[] }: {Data:map[TTC:{Quote:map[USD:{Price:0.04859286}] Symbol:TTC}]}
TRICK, {Data:map[]}
{map[] }: {Data:map[]}
TRON, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TSC:{Quote:map[USD:{Price:3.9207556e-05}] Symbol:TSC}]}
TRKC, {Data:map[]}
{map[] }: {Data:map[]}
TUR, {Data:map[]}
{map[] }: {Data:map[]}
TRANSF, {Data:map[]}
{map[] }: {Data:map[]}
TOT, {Data:map[]}
{map[] }: {Data:map[]}
TRA, {Data:map[]}
TPG, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[]}
TOR, {Data:map[]}
{map[] }: {Data:map[]}
TRBO, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TRC:{Quote:map[USD:{Price:0.02291942}] Symbol:TRC}]}
TP1, {Data:map[]}
{map[] }: {Data:map[]}
TRAP, {Data:map[]}
{map[] }: {Data:map[]}
TODAY, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TOA:{Quote:map[USD:{Price:0.0005485151}] Symbol:TOA}]}
{map[] }: {Data:map[TOKC:{Quote:map[USD:{Price:0.00015677566}] Symbol:TOKC}]}
TOKEN, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TNT:{Quote:map[USD:{Price:0.019999066}] Symbol:TNT}]}
TOOL, {Data:map[]}
{map[] }: {Data:map[]}
TMRW, {Data:map[]}
{map[] }: {Data:map[]}
TLX, {Data:map[]}
{map[] }: {Data:map[]}
TNG, {Data:map[]}
{map[] }: {Data:map[]}
TKN*, {Data:map[]}
{map[] }: {Data:map[]}
TLEX, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TKN:{Quote:map[USD:{Price:0.6307369}] Symbol:TKN}]}
TKLN, {Data:map[]}
{map[] }: {Data:map[]}
TKTX, {Data:map[]}
{map[] }: {Data:map[]}
TLOSH, {Data:map[]}
{map[] }: {Data:map[]}
TLE, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TIT:{Quote:map[USD:{Price:0.0001174994}] Symbol:TIT}]}
TLAC, {Data:map[]}
{map[] }: {Data:map[]}
THOM, {Data:map[]}
{map[] }: {Data:map[]}
THS, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[THETA:{Quote:map[USD:{Price:0.12249295}] Symbol:THETA}]}
THNX, {Data:map[]}
{map[] }: {Data:map[]}
TIO, {Data:map[]}
{map[] }: {Data:map[]}
TIDE, {Data:map[]}
{map[] }: {Data:map[]}
TIA, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TIME:{Quote:map[USD:{Price:1.7763997}] Symbol:TIME}]}
{map[] }: {Data:map[TIE:{Quote:map[USD:{Price:0.020371363}] Symbol:TIE}]}
TES**, {Data:map[]}
{map[] }: {Data:map[]}
TERI, {Data:map[]}
{map[] }: {Data:map[]}
TERRA, {Data:map[]}
{map[] }: {Data:map[]}
TERA, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TFL:{Quote:map[USD:{Price:0.2343622}] Symbol:TFL}]}
TENNET, {Data:map[]}
{map[] }: {Data:map[]}
TFT, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TELL:{Quote:map[USD:{Price:0.00015684238}] Symbol:TELL}]}
TESLA, {Data:map[TESLA:{Quote:map[USD:{Price:0.0009294259}] Symbol:TESLA}]}
{map[] }: {Data:map[TESLA:{Quote:map[USD:{Price:0.0009294259}] Symbol:TESLA}]}
TCR, {Data:map[]}
{map[] }: {Data:map[]}
TEAM, {Data:map[TEAM:{Quote:map[USD:{Price:0.004153649}] Symbol:TEAM}]}
{map[] }: {Data:map[TEAM:{Quote:map[USD:{Price:0.004153649}] Symbol:TEAM}]}
TCOIN, {Data:map[]}
{map[] }: {Data:map[]}
TCX, {Data:map[]}
{map[] }: {Data:map[]}
TECH, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TEK:{Quote:map[USD:{Price:3.9462462e-05}] Symbol:TEK}]}
TEC, {Data:map[]}
{map[] }: {Data:map[]}
TDFB, {Data:map[]}
{map[] }: {Data:map[]}
{map[] }: {Data:map[TCN:{Quote:map[USD:{Price:0.04848876}] Symbol:TCN}]}
TAP, {Data:map[]}
{map[] }: {Data:map[]}
TBCX, {Data:map[]}
{map[] }: {Data:map[]}
TBC, {Data:map[]}
{map[] }: {Data:map[]}
, {Data:map[]}
{map[] }: {Data:map[]}
, {Data:map[]}
{map[] }: {Data:map[]}
NIO*, {Data:map[]}
SPC*, {Data:map[]}
SBD*, {Data:map[]}
RDN*, {Data:map[]}
EDR*, {Data:map[]}
SKRP, {Data:map[]}
TLC, {Data:map[]}
XAU, {Data:map[]}
PAI*, {Data:map[]}
BTCUS, {Data:map[]}
IOT, {Data:map[]}
XBTUSD, {Data:map[]}
EUR, {Data:map[]}
CVCOIN, {Data:map[]}
CBS, {Data:map[]}
MAN*, {Data:map[]}
SMT*, {Data:map[]}
ARC**, {Data:map[]}
WLK, {Data:map[]}
RBM, {Data:map[]}
ATFS, {Data:map[]}
ACT*, {Data:map[]}
GPC, {Data:map[]}
wETT, {Data:map[]}
ZONTO, {Data:map[]}
ZNE, {Data:map[]}
ZOOM, {Data:map[]}
ZXT, {Data:map[]}
XNA, {Data:map[]}
XFCX, {Data:map[]}
XEN, {Data:map[]}
XFB, {Data:map[]}
XDB, {Data:map[]}
XDE2, {Data:map[]}
XDE, {Data:map[]}
XBU, {Data:map[]}
XIN*, {Data:map[]}
XID, {Data:map[]}
XIOS, {Data:map[]}
XFX, {Data:map[]}
XID*, {Data:map[]}
XGTC, {Data:map[]}
XGB, {Data:map[]}
XLTCG, {Data:map[]}
XMS, {Data:map[]}
XMINE, {Data:map[]}
XLTC, {Data:map[]}
ZLQ, {Data:map[]}
ZET2, {Data:map[]}
ZMC, {Data:map[]}
ZGC, {Data:map[]}
XON, {Data:map[]}
XNM, {Data:map[]}
XNX, {Data:map[]}
XOC, {Data:map[]}
XMT, {Data:map[]}
XVE, {Data:map[]}
XNG, {Data:map[]}
XODUS, {Data:map[]}
XOM, {Data:map[]}
XNC, {Data:map[]}
eETT, {Data:map[]}
XPOKE, {Data:map[]}
XPS, {Data:map[]}
XPO, {Data:map[]}
XPT, {Data:map[]}
XPRO, {Data:map[]}
XRB, {Data:map[]}
XSSX, {Data:map[]}
XRL, {Data:map[]}
XSP, {Data:map[]}
XRC, {Data:map[]}
ZIRK, {Data:map[]}
YBC, {Data:map[]}
XZA, {Data:map[]}
YES, {Data:map[]}
YOB2X, {Data:map[]}
XWT, {Data:map[]}
XUP, {Data:map[]}
XTO, {Data:map[]}
XVS, {Data:map[]}
XT, {Data:map[]}
XTP, {Data:map[]}
XSY, {Data:map[]}
XTR, {Data:map[]}
ZBCN, {Data:map[]}
ZDC, {Data:map[]}
YMC, {Data:map[]}
YAY, {Data:map[]}
YAC, {Data:map[]}
ZBC*, {Data:map[]}
ZBC, {Data:map[]}
YYW, {Data:map[]}
YOG, {Data:map[]}
ZECD, {Data:map[]}
