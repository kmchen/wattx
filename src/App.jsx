import React, { Component } from "react";
import ReactDOM from "react-dom";
import { connect } from 'react-redux'
import Fuse from 'fuse.js';

import { Nav } from "./components/Nav";
import { Sort } from "./components/Sort";
import { Results } from "./components/Results";
import { SearchBar } from "./components/SearchBar";
import { PUSH_SEARCH_RESULT } from './actions/actions';

export class AppComponent extends Component {
  constructor() {
    super();
    this.state = {
      searchText: '',
      results: [],
      priceSortOrder: null,
      ratingSortOrder: null,
      priceSortName: 'Price',
      ratingSortName: 'Rating'
    };
  }

  componentDidMount() {
    //if(this.state.results.length === 0) {
      //const specialOffers = this.props.data.filter(x => x.isSpecialOffer);
      //this.setState({results: specialOffers});
    //}
  }

  onSearchTextChange = (event) => {
    this.setState({ searchText: event.target.value });
  }

  onSearch = (event) => {
    if(event.key === 'Enter') {
      if (this.state.searchText === '') {
        return; 
      }
      const fuse = new Fuse(this.props.data, options);
      const searchResults = fuse.search(this.state.searchText);
      this.setState({ results: searchResults});
      this.props.pushSearchResults({keyword: this.state.searchText, result: searchResults})
    }
  }

  onDelete = () => {
    this.setState({searchText: ''});
  }

  onSetResults = (histories) => {
    this.setState({results: histories.result, searchText: histories.keyword});
  }

  sortByPrice = () => {
    const { priceSortOrder } = this.state;
    const newResults = this.state.results.slice();
    const sortedResults = newResults.sort(this.compare('price'));
    if (!priceSortOrder) {
      this.setState({
        priceSortOrder: 'ACS',
        priceSortName: 'Price↑',
        ratingSortOrder: null,
        ratingSortName: 'Rating',
        results: sortedResults
      });
      return
    }
    if (priceSortOrder === 'ACS') {
      this.setState({
        priceSortOrder: 'DESC',
        priceSortName: 'Price↓',
        ratingSortOrder: null,
        ratingSortName: 'Rating',
        results: sortedResults.reverse()
      });
      return
    }
    if (priceSortOrder === 'DESC') {
      this.setState({
        priceSortOrder: 'ACS',
        priceSortName: 'Price↑',
        ratingSortOrder: null,
        ratingSortName: 'Rating',
        results: sortedResults
      });
    }
  }

  sortByRating = () => {
    const { ratingSortOrder } = this.state;
    const newResults = this.state.results.slice();
    const sortedResults = newResults.sort(this.compare('rating'));
    if (!ratingSortOrder) {
      this.setState({
        ratingSortOrder: 'ACS',
        ratingSortName: 'Rating↑',
        priceSortOrder: null,
        priceSortName: 'Price',
        results: sortedResults
      });
      return
    }
    if (ratingSortOrder === 'ACS') {
      this.setState({
        ratingSortOrder: 'DESC',
        ratingSortName: 'Rating↓',
        priceSortOrder: null,
        priceSortName: 'Price',
        results: sortedResults.reverse()
      });
      return
    }
    if (ratingSortOrder === 'DESC') {
      this.setState({
        ratingSortOrder: 'ACS',
        ratingSortName: 'Rating↑',
        priceSortOrder: null,
        priceSortName: 'Price',
        results: sortedResults
      });
    }
  }

  compare = field => {
    return (a, b) => {
      if (a[field] < b[field])
          return -1;
      if (a[field] > b[field])
          return 1;
      return 0;
    };
  }

  render() {
    return (
      <>
        <section className="hero is-fullheight is-default is-bold">
          <Nav />
        </section>
      </>
    );
  }
}

const mapStateToProps = state => {
  return {
    data: state.data,
    histories: state.histories
  }
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    pushSearchResults: results => {
      dispatch({type: PUSH_SEARCH_RESULT, payload: results})
    }
  }
}

const options = {
	shouldSort: true,
	threshold: 0.0,
	location: 0,
	distance: 100,
	maxPatternLength: 32,
	minMatchCharLength: 1,
	keys: [
		'title',
		'price',
		'currency',
		'rating',
		'isSpecialOffer'
	]
};

const App = connect(
  mapStateToProps,
  mapDispatchToProps
)(AppComponent);

export default App;
