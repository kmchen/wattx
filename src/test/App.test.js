import React from 'react';
import { shallow, configure } from 'enzyme';
import { AppComponent } from '../App';
import Adapter from 'enzyme-adapter-react-16';
import { data } from '../../data/data.js';

configure({ adapter: new Adapter() });

describe('App component', () => {
	let appComponent;
  
	beforeEach(() => {
		const props = {
			pushSearchResults: jest.fn(),
	    data: data.tours,
			histories: []
		};
		appComponent = shallow(<AppComponent {...props} />);
	});

	it('should render App component', () => {
		expect(appComponent).toMatchSnapshot();
	});
  
	it('should dislay special offers for the first time', () => {
		const specialOffers = appComponent.state('results');
		specialOffers.forEach(offer => {
			expect(offer.isSpecialOffer).toBeTruthy();
		});
	});
});
