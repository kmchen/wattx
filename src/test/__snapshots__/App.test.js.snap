// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component 1`] = `
<Fragment>
  <Nav />
  <SearchBar
    histories={Array []}
    onChange={[Function]}
    onDelete={[Function]}
    onKeyPress={[Function]}
    onSetResults={[Function]}
    searchText=""
  />
  <Sort
    priceSortName="Price"
    ratingSortName="Rating"
    sortByPrice={[Function]}
    sortByRating={[Function]}
  />
  <Results
    results={
      Array [
        Object {
          "currency": "$",
          "isSpecialOffer": true,
          "price": "41",
          "rating": "4.5",
          "title": "Berlin: 2.5-Hour Boat Tour Along the River Spree",
        },
        Object {
          "currency": "$",
          "isSpecialOffer": true,
          "price": "50",
          "rating": "4.6",
          "title": "Reichstag: Rooftop Coffee Break at KÃ¤fer",
        },
        Object {
          "currency": "$",
          "isSpecialOffer": true,
          "price": "46",
          "rating": "4.9",
          "title": "Skip the Line: Neues Museum Berlin Tickets",
        },
      ]
    }
  />
</Fragment>
`;
